container-api/src/main/java/org/jboss/forge/furnace/repositories/AddonStateRepository.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2012 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Eclipse Public License version 1.0, available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.jboss.forge.furnace.repositories;
+
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.addons.AddonId;
+import org.jboss.forge.furnace.versions.Version;
+
+import java.util.List;
+
+/**
+ * Used to perform {@link Addon} registration operations.
+ * 
+ * @author <a href="mailto:lincolnbaxter@gmail.com">Lincoln Baxter, III</a>
+ * @author <a href="mailto:koen.aers@gmail.com">Koen Aers</a>
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public interface AddonStateRepository
```

Comment:
Pushed. Just to keep track of this naming - I still disagree with it, but you're the owner of the code base and it's up to you to do some final decisions about the naming :)

Especially it looks weird here:

```
public interface AddonRepository extends AddonRepositoryStorageStrategy, AddonRepositoryStateStrategy
```

AddonRepository interface (which used to be a single thing) was split into the two, but now these two things are not repositories somehow anymore :D

But I really want to get this change in upstream so I'm fine with strategies :)